File: DataCommon/DataCommon/Entities/Product.cs
using DataCommon.Request;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DataCommon.Entities
{
    public class Product : BaseEntity
    {
        public string? Name { get; set; }
        public string? Description { get; set; }
        public decimal? Price { get; set; }
        public string? ImageURL { get; set; }
        public string? CreatedDate { get; set; }
        public string? UpdatedDate { get; set; }
        public decimal? RatingPoint { get; set; }
        public Guid CategoryId { get; set; }
        public Category? Category { get; set; }

        public Product()
        {

        }

        public Product(ProductRequestModel product, Category category)
        {
            Id = product.Id;
            Name = product.Name;
            Description = product.Description;
            Price = product.Price;
            ImageURL = product.ImageURL;
            CreatedDate = product.CreatedDate;
            UpdatedDate = product.UpdatedDate;
            CategoryId = product.CategoryId;
            Category = category;
        }
    }
}

File: ShopooAdminApp/src/app/models/product/response/product.ts
import { CategoryModel } from "../../category/response/category";

export class ProductModel {
    public name!: string | null;
    public description!: string | null;
    public price!: number | null;
    public imageURL!: string | null;
    public createdDate!: string | null;
    public updatedDate!: string | null;
    public ratingPoint!: number | null;
    public categoryId!: string;
    public category!: CategoryModel | null;
}